src/data/slice.ts

```diff
@@ -0,0 +1,220 @@
+import {
+	createSlice,
+	createEntityAdapter,
+	PayloadAction,
+} from '@reduxjs/toolkit';
+
+import { CodemodEntry } from '../codemods/types';
+import { ExecutionError } from '../errors/types';
+import { PersistedCase, PersistedJob } from '../persistedState/codecs';
+
+const SLICE_KEY = 'root';
+
+type CodemodDiscoveryState = Readonly<{
+	openedCodemodHashDigests: ReadonlyArray<string> | null;
+	focusedCodemodHashDigest: string | null;
+	executionPaths: Record<string, string>;
+	visible: boolean;
+}>;
+
+type CodemodRunsState = Readonly<{
+	selectedCaseHash: string | null;
+	visible: boolean;
+}>;
+
+type ChangeExplorerState = Readonly<{
+	visible: boolean;
+	focusedFileExplorerNodeId: string | null;
+	openedFileExplorerNodeIds: ReadonlyArray<string> | null;
+}>;
+
+type CommunityState = Readonly<{
+	visible: boolean;
+}>;
+
+type State = {
```

Comment:
I recall you wanted this to become normalized?

Reply:
normalization is handled by adapters, i use it for entity collection only